lib/utils/format.js

```diff
@@ -0,0 +1,97 @@
+const { formatWithOptions: baseFormatWithOptions, inspect } = require('util')
+
+const CUSTOM_INSPECT = Symbol('npm.display.original.util.inspect.custom')
+
+// These are most assuredly not a mistake
+// https://eslint.org/docs/latest/rules/no-control-regex
+// \x00 through \x1f, \x7f through \x9f, not including \x09 \x0a \x0b \x0d
+/* eslint-disable-next-line no-control-regex */
+const HAS_C01 = /[\x00-\x08\x0c\x0e-\x1f\x7f-\x9f]/
+
+// Allows everything up to '[38;5;255m' in 8 bit notation
+const ALLOWED_SGR = /^\[[0-9;]{0,8}m/
+
+// '[38;5;255m'.length
+const SGR_MAX_LEN = 10
+
+// Strips all ANSI C0 and C1 control characters (except for SGR up to 8 bit)
+function STRIP_C01 (str) {
+  if (!HAS_C01.test(str)) {
+    return str
+  }
+  let result = ''
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i]
+    const code = char.charCodeAt(0)
+    if (!HAS_C01.test(char)) {
+      // Most characters are in this set so continue early if we can
+      result = `${result}${char}`
+    } else if (code === 27 && ALLOWED_SGR.test(str.slice(i + 1, i + SGR_MAX_LEN + 1))) {
+      // \x1b with allowed SGR
+      result = `${result}\x1b`
+    } else if (code <= 31) {
+      // escape all other C0 control characters besides \x7f
+      result = `${result}^${String.fromCharCode(code + 64)}`
+    } else {
+      // hasC01 ensures this is now a C1 control character or \x7f
+      result = `${result}^${String.fromCharCode(code - 64)}`
+    }
+  }
+  return result
+}
+
+const createCleaner = (fn) => (output) => {
```

Comment:
You're gonna have to walk me through this change.

Reply:
I was just having fun here lol. I think this should be rolled back or it might be possible to do this in a single pass now.

Reply:
Since we now pass everything through `util.format` first it looks like we can strip control characters on the returned string and now have to worry about `util.inspect.custom` anymore.

I'm relying heavily on the existing tests here but they all pass.

Reply:
You may be right.  The reason we had to do it this way was because we did not control the npmlog code from npm itself, and it was ultimately passing things to `console.log` that we had no control over.

Reply:
I'm gonna leave this thread unresolved until I can review that use case specifically.